### PR TITLE
Support different inputs for ROC

### LINKED ISSUE #98: Multiclass ROC curve
ROC curve is commonly used to compare the performance of models. It is usually used in binary classification, but it can also be used in multiclass classification using averaging methods.

When running `plot.roc(y_test, prediction1)` on more than 2 classes it fails, we should add support to it:

`ValueError: multiclass format is not supported`

I tried it for ROC curve, but this issue also applies to PR curve. (`plot.precision_recall(y_test, prediction1`)

### PR DESCRIPTION
**Pull request by yafimvo**
## Describe your changes
Added support for 3 different inputs for ROC curve: lables/classes array, onehot encoded array, and array of values.

```
array([0, 1, 2])

array([[0, 1, 0], 
        [1, 0, 0], 
        [0, 0, 1]])

array([[0.1, 0.8, 0.1], 
        [0.8, 0.1, 0.1], 
        [0.1, 0.1, 0.8]])
```

## Issue ticket number and link
Closes #98  

## Checklist before requesting a review
- [X] I have performed a self-review of my code
- [X] I have added thorough tests (when necessary).
- [X] I have added the right documentation (when needed). Product update? If yes, write one line about this update.

----
:books: Documentation preview :books:: https://sklearn-evaluation--221.org.readthedocs.build/en/221/

### CODE REVIEW

**Comment by yafimvo** [comment]
@edublancas  @idomic 
I added an example to [classifications](https://sklearn-evaluation--221.org.readthedocs.build/en/221/quickstart/classification.html#evaluate-our-model). 

I think it's not the best example for ROC since the accuracy is almost 100% and the curves are "too good"

image

**Review by idomic — APPROVED** [review]
Looking good, just rebase on main and let's merge this.

File: docs/quickstart/classification.md [hunk 1]
```diff
@@ -99,6 +99,27 @@ plot.confusion_matrix(y_test, y_pred_dt)
 plot.feature_importances(dtc, top_n=5, feature_names=list(dtc.feature_names_in_))
 ```
 
+Use the predicted labels or classes to plot an ROC curve.
+We support 3 main formats that allow you to quickly generate a single or a multi classification ROC curve.
+
+```
```
Comment (edublancas) [inline docs/quickstart/classification.md:105]: let's move this to the docstring. and clarify that the first two formats are for the y_true argument and the second format is for y_score

File: docs/quickstart/classification.md [hunk 2]
```diff
@@ -99,6 +99,27 @@ plot.confusion_matrix(y_test, y_pred_dt)
...
+
+array([[0.1, 0.8, 0.1], 
+        [0.8, 0.1, 0.1], 
+        [0.1, 0.1, 0.8]])
+```
+
+For this example we will plot a multi classification ROC using the predicted classes 
+
+```{code-cell} ipython3
+plot.roc(y_test, y_pred_dt)
```
Comment (edublancas) [inline docs/quickstart/classification.md:120]: there's an error here, y_pred_dt is created with `y_pred_dt = dt_model.predict(X_test)`

but it should be `y_pred_dt = dt_model.predict_proba(X_test)`

note that some plots like confusion matrix need the output of `predict`, and others like roc need `predict_proba`

**Review by edublancas — CHANGES_REQUESTED** [review]

**Comment by edublancas** [comment]
found a minor issue

**Comment by yafimvo** [comment]
@edublancas @idomic 
This PR should be closed since we[ took another approach](https://github.com/ploomber/sklearn-evaluation/issues/98#issuecomment-1387078829) where ROC inputs should be "scores" like for `y_score` and "classes"/"one hot" for `y_test`.

**Pull request closed without merging**